From 31f0ea59ba6a1d4f494a7cb2b6d9403e5140a563 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Tue, 27 Nov 2018 16:56:27 +0800
Subject: [PATCH] usb: gadget: f_uvc: increase the data buf of uvc_request_data

The rockchip IQ tool uses the data buf of the uvc_request_data
to send vendor specific uvc control data to uvc host, or receive
data from uvc host. The max data payload maybe 4096 bytes, so we
increase the data buf of uvc_request_data to 4096, and also change
the UVC_MAX_REQUEST_SIZE to 4100 correspondingly.

Change-Id: I9820a5f531a655eb53232096fad131ad7a05e349
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/gadget/function/uvc.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/uvc.h b/drivers/usb/gadget/function/uvc.h
index 7d3bb6272e06..1134921ad587 100644
--- a/drivers/usb/gadget/function/uvc.h
+++ b/drivers/usb/gadget/function/uvc.h
@@ -29,7 +29,11 @@
 struct uvc_request_data
 {
 	__s32 length;
+#ifdef	CONFIG_USB_CONFIGFS_F_UVC_ROCKCHIP
+	__u8 data[4096];
+#else
 	__u8 data[60];
+#endif
 };
 
 struct uvc_event
@@ -97,8 +101,12 @@ extern unsigned int uvc_gadget_trace_param;
  */
 
 #define UVC_NUM_REQUESTS			4
-#define UVC_MAX_REQUEST_SIZE			64
 #define UVC_MAX_EVENTS				4
+#ifdef	CONFIG_USB_CONFIGFS_F_UVC_ROCKCHIP
+#define UVC_MAX_REQUEST_SIZE			4100
+#else
+#define UVC_MAX_REQUEST_SIZE			64
+#endif
 
 /* ------------------------------------------------------------------------
  * Structures
-- 
2.35.3

